src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ServiceProviderIsServiceSpecificationTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Specification
+{
+    public abstract class ServiceProviderIsServiceSpecificationTests
```

Comment:
We only have the instrumentation to run spec tests for 3rd party containers enabled for the main test class.

Reply:
I started that way but moved away from it because it wasn't very clean. I also tried adding a virtual property to enable running these tests but there's no first class xunit way to skip based on that. Do you have a cleaner idea on how to suppress failures?

Reply:
https://github.com/dotnet/runtime/blob/davidfowl/serviceprovider-isservice/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/StructureMap.cs#L11 ??

Reply:
I think if we need to do this then the default base class needs to have this feature. This feature is completely optional and if we have it as part of the spec tests then we need to built this skippibility (I made that word up) as part of the default experience.

Reply:
can you use an `if` inside the and a virtual property to skip?
```
public virtual SupportsIsService {get;} = true;


public void IsServiceWorks()
{
if (SupportsIsService)
{
// test code.

}
}
```

Reply:
This is what I did originally but it felt so bad. You think its better to do it this way?

Reply:
It gives us coverage and 3rd parties a way to discover new features when updating the spec test package.

Reply:
So is it for us or for 3rd parties? How will they discover it if all of the tests pass and the virtual returns false by default?

Reply:
>So is it for us or for 3rd partie

Both.

>How will they discover it if all of the tests pass and the virtual returns false by default?

We'll turn it on by default.

Reply:
Done